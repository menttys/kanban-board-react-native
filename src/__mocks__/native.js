export const mockNavigate = jest.fn();
export const mockGoBack = jest.fn();

export const useNavigation = () => ({
  navigate: mockNavigate,
  goBack: mockGoBack,
});
