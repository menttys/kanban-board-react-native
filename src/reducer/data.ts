export const mockData = {
  lastIndice: 3,
  boards: {
    ['0']: {
      name: 'Backlog',
      tickets: [
        {
          id: 'A1',
          title: 'Chewbacca',
          description:
            'This is the first boy, This is the first boy This is the first boy This is the first boy This is the first boy This is the first boy This is the first boy',
          tag: 'Star Wars',
          link: 'https://www.google.com',
        },
      ],
    },
    ['1']: {
      name: 'To Do',
      tickets: [
        {
          id: 'A2',
          title: 'Row 5',
          description:
            'This is the first boy, This is the first boy This is the first boy This is the first boy This is the first boy This is the first boy This is the first boy',
          tag: 'Star Wars',
          link: 'https://www.google.com',
        },
        {
          id: 'A3',
          title: 'Row 6 ',
          description:
            'This is the first boy, This is the first boy This is the first boy This is the first boy This is the first boy This is the first boy This is the first boy',
          tag: 'Marvel',
          link: 'https://www.google.com',
        },
      ],
    },
    ['2']: {
      name: 'In Development',
      tickets: [],
    },
    ['3']: {
      name: 'Completed',
      tickets: [],
    },
  },
};
