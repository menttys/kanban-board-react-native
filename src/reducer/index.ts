export {
  initialState,
  reducer,
  TICKET_ACTIONS,
  TICKETS_ACTION_TYPE,
  TICKET_STATE,
  STEPS_STATE,
  BOARD_STATE,
  STATE,
} from './ticketsReducer';
