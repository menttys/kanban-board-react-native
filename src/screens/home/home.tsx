import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  ScrollView,
  TouchableOpacity,
  TextInput,
} from 'react-native';
import { useNavigation } from '@react-navigation/native';
import {
  DraxProvider,
  DraxView,
  DraxSnapbackTargetPreset,
} from 'react-native-drax';

import {
  AppNavigatorRoutes,
  AppNavigatorNavigationProp,
} from '@app/navigator/AppNavigator';
import Card from '@app/components/Card';
import Picker from '@app/components/Picker';
import Icon from '@app/components/Icon';
import {
  initialState,
  reducer,
  TICKET_ACTIONS,
  STEPS_STATE,
  BOARD_STATE,
} from '@app/reducer';

export enum Stages {
  BACKLOG = 'Backlog',
  TODO = 'To Do',
  IN_DEVELOPMENT = 'In Development',
  COMPLETED = 'Completed',
}

const stagesColors = {
  [Stages.BACKLOG]: '#1B5A7C',
  [Stages.TODO]: '#106254',
  [Stages.IN_DEVELOPMENT]: '#71441B',
  [Stages.COMPLETED]: '#54127D',
};

const COLUMN_WIDTH = 250;

/**
 * @function @Home
 **/

export const Home = () => {
  const navigation = useNavigation<AppNavigatorNavigationProp>();

  const [state, dispatch] = React.useReducer(reducer, initialState);
  const [selectedTickets, setselectedTickets] = React.useState(state);

  const {
    add_ticket_button,
    empty_column,
    empty_column_text,
    header_container,
    home_container,
    main_container,
    step,
    step_header,
    search_container,
    search_icon,
    search_text,
  } = styles;

  const handleSearch = (event: any) => {
    // will search only when lenght larger than 3
    if (event.nativeEvent.text.length > 3) {
      let newBoard = {};
      Object.keys(selectedTickets.boards as BOARD_STATE).forEach((step) => {
        const filteredTickets =
          state.boards[step].tickets.filter(
            (ticket: {
              title: string;
              tag: string;
              description: string;
              link: string;
            }) =>
              ticket.title.includes(event.nativeEvent.text) ||
              ticket.tag.includes(event.nativeEvent.text) ||
              ticket.description.includes(event.nativeEvent.text) ||
              ticket.link.includes(event.nativeEvent.text),
          ) || [];

        newBoard = {
          ...newBoard,
          [step]: {
            name: state.boards[step].name,
            tickets: filteredTickets,
          },
        };
      });
      const newState = {
        ...state,
        boards: newBoard,
      };

      setselectedTickets(() => newState);
    }
    // clean up the search
    if (event.nativeEvent.text.length === 3) {
      setselectedTickets(() => state);
    }
  };

  return (
    <View style={main_container}>
      <View style={{ ...header_container, zIndex: 100, position: 'relative' }}>
        <View style={search_container}>
          <View style={search_icon}>
            <Icon name="Search" size="xlg" />
          </View>
          <TextInput
            style={search_text}
            placeholderTextColor="#222"
            placeholder="Search this board"
            onChange={(event) => handleSearch(event)}
          />
        </View>
        <Picker />
      </View>

      <ScrollView horizontal style={{ zIndex: 0 }}>
        <DraxProvider>
          <View style={home_container}>
            {selectedTickets &&
              Object.values(selectedTickets.boards as BOARD_STATE).map(
                (data: STEPS_STATE, columnIndex) => {
                  return (
                    <ScrollView
                      key={`${data.name}_${columnIndex}`}
                      testID="boards-container"
                    >
                      <DraxView
                        style={{
                          ...step,
                          backgroundColor: stagesColors[data.name as Stages],
                        }}
                        onReceiveDragDrop={({ dragged: { payload } }: any) => {
                          dispatch({
                            type: TICKET_ACTIONS.MOVE,
                            payload: {
                              ticketId: payload.ticketId,
                              newColumn: `${columnIndex}`,
                              actualColumn: payload.actualColumn,
                            },
                          });
                          return DraxSnapbackTargetPreset.None;
                        }}
                        receptive={true}
                      >
                        <Text style={step_header}>{data.name}</Text>
                        {!data.tickets.length ? (
                          <View style={empty_column}>
                            <Text
                              style={{
                                ...empty_column_text,
                                color: stagesColors[data.name as Stages],
                              }}
                            >
                              Drag and drop here
                            </Text>
                          </View>
                        ) : (
                          data.tickets.map((ticket, ticketIndex) => (
                            <Card
                              key={`${ticket.id}${ticketIndex}`}
                              ticket={ticket}
                              column={columnIndex}
                              testID={`board-${columnIndex}`}
                            />
                          ))
                        )}

                        {data.name === Stages.BACKLOG && (
                          <TouchableOpacity
                            onPress={() =>
                              navigation.navigate(AppNavigatorRoutes.ADD_CARD, {
                                dispatch: dispatch,
                                state: state,
                              })
                            }
                            style={{
                              paddingVertical: 12,
                            }}
                          >
                            <Text style={add_ticket_button}>+ Add Card</Text>
                          </TouchableOpacity>
                        )}
                      </DraxView>
                    </ScrollView>
                  );
                },
              )}
          </View>
        </DraxProvider>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  main_container: {
    flex: 1,
    backgroundColor: 'white',
  },
  home_container: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
    paddingVertical: 12,
    paddingHorizontal: 8,
  },
  header_container: {
    marginBottom: 20,
    flexDirection: 'row',
    paddingTop: 8,
    paddingBottom: 24,
    paddingHorizontal: 16,
    backgroundColor: '#fcfdfc',
    shadowColor: '#888',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.5,
    shadowRadius: 3,
    elevation: 5,
    alignItems: 'center',
  },
  step: {
    width: COLUMN_WIDTH,
    marginHorizontal: 8,
    paddingHorizontal: 12,
    padding: 16,
    borderRadius: 4,
    backgroundColor: 'grey',
  },
  step_header: {
    color: 'white',
    fontFamily: 'nunito',
    fontSize: 16,
    marginBottom: 16,
  },
  empty_column: {
    backgroundColor: 'white',
    padding: 16,
    borderRadius: 4,
  },
  empty_column_text: {
    fontSize: 18,
  },
  add_ticket_button: {
    color: 'white',
    fontSize: 16,
    width: '100%',
  },
  search_container: {
    borderColor: '#aaa',
    borderWidth: 1,
    flex: 3,
    borderRadius: 36,
    flexDirection: 'row',
    alignItems: 'center',
  },
  search_icon: {
    marginLeft: 12,
  },
  search_text: {
    fontSize: 18,
    paddingVertical: 18,
    paddingLeft: 8,
  },
});
