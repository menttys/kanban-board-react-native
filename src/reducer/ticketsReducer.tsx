import { mockData } from './data';

export enum TICKET_ACTIONS {
  ADD = 'add',
  DELETE = 'delete',
  MOVE = 'move',
}

export type TICKET_STATE = {
  id: string;
  title: string;
  tag: string;
  description: string;
  link: string;
};
export type STEPS_STATE = {
  name: string;
  tickets: TICKET_STATE[];
};
export type BOARD_STATE = Record<string, STEPS_STATE>;
export type STATE = {
  lastIndice: number;
  boards: BOARD_STATE;
};

export const initialState: STATE = mockData;

type PAYLOAD_TYPE = {};

export type TICKETS_ACTION_TYPE =
  | {
      type: TICKET_ACTIONS.ADD;
      payload: {
        tag: string;
        cardTitle: string;
        cardDescription: string;
        link: string;
      };
    }
  | { type: TICKET_ACTIONS.DELETE; payload: PAYLOAD_TYPE }
  | {
      type: TICKET_ACTIONS.MOVE;
      payload: {
        ticketId: string;
        newColumn: string;
        actualColumn: string;
      };
    };

export function reducer(state: STATE, action: TICKETS_ACTION_TYPE): STATE {
  switch (action.type) {
    case TICKET_ACTIONS.ADD:
      const { tag, cardTitle, cardDescription, link } = action.payload;
      const addState: STATE = { ...state, lastIndice: +state.lastIndice + 1 };
      addState.boards[0].tickets.push({
        id: `A${+state.lastIndice + 1}`,
        title: cardTitle,
        description: cardDescription,
        tag: tag,
        link: link,
      });
      return {
        ...addState,
      };

    case TICKET_ACTIONS.DELETE:
      return {
        ...state,
      };

    case TICKET_ACTIONS.MOVE:
      const { ticketId, newColumn, actualColumn } = action.payload;
      const newState = { ...state };
      let savedTicket: TICKET_STATE | null = null;
      const removeTicket = newState.boards[actualColumn].tickets.filter(
        (ticket) => {
          if (ticket.id !== ticketId) {
            return ticket;
          }
          savedTicket = ticket;
        },
      );
      newState.boards[actualColumn].tickets = removeTicket;
      if (savedTicket) {
        newState.boards[newColumn].tickets = [
          ...newState.boards[newColumn].tickets,
          savedTicket,
        ];
      }

      return {
        ...newState,
      };
    default:
      throw new Error();
  }
}
