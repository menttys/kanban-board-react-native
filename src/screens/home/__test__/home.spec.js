import React from 'react';
import { render, screen } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import { mockData } from '@app/reducer/data';
import { Home } from '../home';

const numberOfBoards = Object.keys(mockData.boards).length;

const renderContainer = () =>
  render(
    <NavigationContainer>
      <Home />
    </NavigationContainer>,
  );

describe('<Home> Screen', () => {
  it('should render the columns steps', async () => {
    renderContainer();

    const boards = screen.getAllByTestId('boards-container');
    expect(boards.length).toBe(numberOfBoards);
  });

  it('should render the right number of ticket to each column', async () => {
    renderContainer();

    expect(screen.getAllByTestId('board-0').length).toBe(
      mockData.boards[0].tickets.length,
    );
    expect(screen.getAllByTestId('board-1').length).toBe(
      mockData.boards[1].tickets.length,
    );
  });

  it('home snapshot', () => {
    const { toJSON } = renderContainer();

    expect(toJSON()).toMatchSnapshot();
  });
});
