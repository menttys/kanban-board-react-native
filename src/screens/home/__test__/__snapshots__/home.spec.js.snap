// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home> Screen home snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fcfdfc",
        "elevation": 5,
        "flexDirection": "row",
        "marginBottom": 20,
        "paddingBottom": 24,
        "paddingHorizontal": 16,
        "paddingTop": 8,
        "position": "relative",
        "shadowColor": "#888",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 3,
        "zIndex": 100,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#aaa",
          "borderRadius": 36,
          "borderWidth": 1,
          "flex": 3,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "marginLeft": 12,
          }
        }
      >
        <SvgMock
          height={32}
          width={32}
        />
      </View>
      <TextInput
        onChange={[Function]}
        placeholder="Search this board"
        placeholderTextColor="#222"
        style={
          Object {
            "fontSize": 18,
            "paddingLeft": 8,
            "paddingVertical": 18,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "paddingHorizontal": 12,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
        Tag
      </Text>
    </View>
  </View>
  <RCTScrollView
    horizontal={true}
    style={
      Object {
        "zIndex": 0,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "padding": 24,
              "paddingHorizontal": 8,
            }
          }
        >
          <View
            testID="boards-container"
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              handlerTag={15}
              handlerType="LongPressGestureHandler"
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              onLayout={[Function]}
              style={
                Object {
                  "backgroundColor": "#1B5A7C",
                  "borderRadius": 4,
                  "marginHorizontal": 8,
                  "padding": 16,
                  "paddingHorizontal": 12,
                  "width": 250,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginBottom": 16,
                  }
                }
              >
                Backlog
              </Text>
              <View
                collapsable={false}
                forwardedRef={[Function]}
                handlerTag={16}
                handlerType="LongPressGestureHandler"
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                onLayout={[Function]}
                style={
                  Object {
                    "alignItems": "flex-start",
                    "backgroundColor": "white",
                    "borderRadius": 4,
                    "marginBottom": 12,
                    "padding": 12,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#222",
                      "borderRadius": 4,
                      "marginBottom": 16,
                      "padding": 8,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontFamily": "Nunito",
                        "fontSize": 14,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Star Wars
                  </Text>
                </View>
                <View
                  testID="board-0"
                >
                  <Text
                    style={
                      Object {
                        "color": "black",
                        "fontFamily": "Nunito",
                        "fontSize": 14,
                        "fontWeight": "bold",
                        "lineHeight": 18,
                        "marginBottom": 12,
                      }
                    }
                  >
                    Chewbacca
                  </Text>
                  <Text
                    numberOfLines={2}
                    style={
                      Object {
                        "color": "#111",
                        "fontFamily": "Nunito",
                        "fontSize": 14,
                        "lineHeight": 18,
                        "marginBottom": 12,
                      }
                    }
                  >
                    This is the first boy, This is the first boy This is the first boy This is the first boy This is the first boy This is the first boy This is the first boy
                  </Text>
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    testID="card-link"
                  >
                    <Text
                      style={
                        Object {
                          "color": "666",
                          "fontFamily": "Nunito",
                          "textDecorationLine": "underline",
                        }
                      }
                    >
                      Document link
                    </Text>
                  </View>
                </View>
              </View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                    "paddingVertical": 12,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "width": "100%",
                    }
                  }
                >
                  + Add Card
                </Text>
              </View>
            </View>
          </View>
          <View
            testID="boards-container"
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              handlerTag={17}
              handlerType="LongPressGestureHandler"
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              onLayout={[Function]}
              style={
                Object {
                  "backgroundColor": "#106254",
                  "borderRadius": 4,
                  "marginHorizontal": 8,
                  "padding": 16,
                  "paddingHorizontal": 12,
                  "width": 250,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginBottom": 16,
                  }
                }
              >
                To Do
              </Text>
              <View
                collapsable={false}
                forwardedRef={[Function]}
                handlerTag={18}
                handlerType="LongPressGestureHandler"
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                onLayout={[Function]}
                style={
                  Object {
                    "alignItems": "flex-start",
                    "backgroundColor": "white",
                    "borderRadius": 4,
                    "marginBottom": 12,
                    "padding": 12,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#222",
                      "borderRadius": 4,
                      "marginBottom": 16,
                      "padding": 8,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontFamily": "Nunito",
                        "fontSize": 14,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Star Wars
                  </Text>
                </View>
                <View
                  testID="board-1"
                >
                  <Text
                    style={
                      Object {
                        "color": "black",
                        "fontFamily": "Nunito",
                        "fontSize": 14,
                        "fontWeight": "bold",
                        "lineHeight": 18,
                        "marginBottom": 12,
                      }
                    }
                  >
                    Row 5
                  </Text>
                  <Text
                    numberOfLines={2}
                    style={
                      Object {
                        "color": "#111",
                        "fontFamily": "Nunito",
                        "fontSize": 14,
                        "lineHeight": 18,
                        "marginBottom": 12,
                      }
                    }
                  >
                    This is the first boy, This is the first boy This is the first boy This is the first boy This is the first boy This is the first boy This is the first boy
                  </Text>
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    testID="card-link"
                  >
                    <Text
                      style={
                        Object {
                          "color": "666",
                          "fontFamily": "Nunito",
                          "textDecorationLine": "underline",
                        }
                      }
                    >
                      Document link
                    </Text>
                  </View>
                </View>
              </View>
              <View
                collapsable={false}
                forwardedRef={[Function]}
                handlerTag={19}
                handlerType="LongPressGestureHandler"
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                onLayout={[Function]}
                style={
                  Object {
                    "alignItems": "flex-start",
                    "backgroundColor": "white",
                    "borderRadius": 4,
                    "marginBottom": 12,
                    "padding": 12,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#222",
                      "borderRadius": 4,
                      "marginBottom": 16,
                      "padding": 8,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontFamily": "Nunito",
                        "fontSize": 14,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Marvel
                  </Text>
                </View>
                <View
                  testID="board-1"
                >
                  <Text
                    style={
                      Object {
                        "color": "black",
                        "fontFamily": "Nunito",
                        "fontSize": 14,
                        "fontWeight": "bold",
                        "lineHeight": 18,
                        "marginBottom": 12,
                      }
                    }
                  >
                    Row 6 
                  </Text>
                  <Text
                    numberOfLines={2}
                    style={
                      Object {
                        "color": "#111",
                        "fontFamily": "Nunito",
                        "fontSize": 14,
                        "lineHeight": 18,
                        "marginBottom": 12,
                      }
                    }
                  >
                    This is the first boy, This is the first boy This is the first boy This is the first boy This is the first boy This is the first boy This is the first boy
                  </Text>
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    testID="card-link"
                  >
                    <Text
                      style={
                        Object {
                          "color": "666",
                          "fontFamily": "Nunito",
                          "textDecorationLine": "underline",
                        }
                      }
                    >
                      Document link
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            testID="boards-container"
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              handlerTag={20}
              handlerType="LongPressGestureHandler"
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              onLayout={[Function]}
              style={
                Object {
                  "backgroundColor": "#71441B",
                  "borderRadius": 4,
                  "marginHorizontal": 8,
                  "padding": 16,
                  "paddingHorizontal": 12,
                  "width": 250,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginBottom": 16,
                  }
                }
              >
                In Development
              </Text>
              <View
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderRadius": 4,
                    "padding": 16,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#71441B",
                      "fontSize": 18,
                    }
                  }
                >
                  Drag and drop here
                </Text>
              </View>
            </View>
          </View>
          <View
            testID="boards-container"
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              handlerTag={21}
              handlerType="LongPressGestureHandler"
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              onLayout={[Function]}
              style={
                Object {
                  "backgroundColor": "#54127D",
                  "borderRadius": 4,
                  "marginHorizontal": 8,
                  "padding": 16,
                  "paddingHorizontal": 12,
                  "width": 250,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginBottom": 16,
                  }
                }
              >
                Completed
              </Text>
              <View
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderRadius": 4,
                    "padding": 16,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#54127D",
                      "fontSize": 18,
                    }
                  }
                >
                  Drag and drop here
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
